xwiki-platform-core/xwiki-platform-skin/xwiki-platform-skin-ui/src/main/resources/SkinsCode/XWikiSkinsSheet.xml

```diff
@@ -80,7 +80,7 @@
   #set ($class = $doc.getObject($className).xWikiClass)
   #foreach ($prop in $class.properties)
     #if ($prop.classType != 'TextArea')
-    ; $services.rendering.escape($prop.prettyName, 'xwiki/2.1')
+    ; {{html}}&lt;label for="${className}_0_$escapetool.xml($prop.name)"&gt;$escapetool.xml($prop.prettyName)&lt;/label&gt;{{/html}}
```

Comment:
It's not necessarily the object at index 0 is it? Chances are low but we could imagine that the retrieved object has another index no?

Reply:
I just tried using the `$prop.number` attribute to fill up this id.
Seems like the number used in the ids is not the same, and so it doesn't match.
![image](https://github.com/xwiki/xwiki-platform/assets/28761965/f2616089-addb-46e0-b85d-281909d64ca3)
From what I could see in this form, all properties used `0` (because they were the only ones with their name, no duplicate names? ).
I'll be leaving the PR as is for now.

Reply:
> I just tried using the $prop.number attribute to fill up this id.

Yes it won't work with `$prop.number`: as the name suggest it's the index of the property. Here the 0 is the index of the `xobject` itself.

Reply:
Addressed in https://github.com/xwiki/xwiki-platform/pull/2642/commits/afe5ee95e8f445fff2d8b56c0f551d70b615f6fe :+1: 

Here is the result with the updated template:
![21521-wihtoutHardcodedNumber](https://github.com/xwiki/xwiki-platform/assets/28761965/778e279f-2e2f-4de2-970c-4df6e212c4d6)
As we can see, all attribute values reference to the right nodes.